### PR TITLE
Working  VLM support for SFT through VERL for 1 batch size

### PR DESCRIPTION
**Pull request by RishuG-work**
### What does this PR do?

> Added VLM support for SFT using Verl library

### CODE REVIEW

File: orby/scripts/eval_screenspot_sft.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
+set -e
```
Comment (ligang-orby) [inline orby/scripts/eval_screenspot_sft.sh:1]: This file seems not modified for SFT? It seems we only need to change the screenspot data generation for SFT eval? If yes, how about adding a new dataset version called "screenspot_sft" below, which then calls python convert_screenspot_sft.py to generate the dataset?

We can modify the original file instead and no need to make a new copy.

File: orby/trainer/fsdp_sft_trainer.py [hunk 2]
```diff
@@ -46,11 +46,10 @@
 from transformers import AutoConfig, AutoModelForCausalLM, PreTrainedModel, Qwen2_5_VLForConditionalGeneration
 from torchdata.stateful_dataloader import StatefulDataLoader
 
-from verl.trainer.main_ppo import create_rl_dataset
 import verl.utils.hdfs_io as hdfs_io
 from verl.utils.dataset import SFTDataset
-from verl.utils.dataset.multiturn_sft_dataset import MultiTurnSFTDataset
-from verl.utils.dataset.rl_dataset import collate_fn
+#from verl.utils.dataset.multiturn_sft_dataset import MultiTurnSFTDataset
```
Comment (ligang-orby) [inline orby/trainer/fsdp_sft_trainer.py:51]: remove commented codes

File: orby/trainer/fsdp_sft_trainer.py [hunk 3]
```diff
@@ -76,6 +75,40 @@
 logger.setLevel(os.getenv("VERL_SFT_LOGGING_LEVEL", "WARN"))
 
 
+def create_rl_dataset(data_paths, data_config, tokenizer, processor):
```
Comment (ligang-orby) [inline orby/trainer/fsdp_sft_trainer.py:78]: let's rename this to create_sft_dataset()

File: orby/trainer/fsdp_sft_trainer.py [hunk 4]
```diff
@@ -76,6 +75,40 @@
...
+        data_config: The data config.
+        tokenizer (Tokenizer): The tokenizer.
+        processor (Processor): The processor.
+
+    Returns:
+        dataset (Dataset): The dataset.
+    """
+    from torch.utils.data import Dataset
+
+    from verl.utils.dataset.rl_dataset import RLHFDataset
```
Comment (ligang-orby) [inline orby/trainer/fsdp_sft_trainer.py:91]: need to import from orby?

File: orby/trainer/fsdp_sft_trainer.py [hunk 5]
```diff
@@ -357,25 +391,42 @@ def _compute_loss_and_backward(self, batch, do_backward=True):
...
+                self.tokenizer.encode("<|im_start|>assistant", add_special_tokens=False), 
+                device=attention_mask.device
+            )
+
+            assistant_len = len(assistant_token_ids)
+            prompt_end_position = torch.zeros(batch_size, device=attention_mask.device, dtype=torch.long)
+
+            # Slide a window across each position to find where assistant tokens start
+            # We only need to check positions where the full assistant sequence could fit
+            for i in range(seq_len - assistant_len + 1):
```
Comment (ligang-orby) [inline orby/trainer/fsdp_sft_trainer.py:406]: The for loop can be quite inefficient -- let's think of better way as TODO item, one option is to create loss mask in the data generation stage.

**Review by ligang-orby — COMMENTED** [review]
Thanks for the PR! Generally lgtm

**Review by ligang-orby — APPROVED** [review]
